simulation_tutorial_2.py failure

**Operating System and version**
Builder `bbq`

**Traceback and/or steps to reproduce**

https://jenkins-ci.tribler.org/job/ipv8/job/validate_documentation/85/console

```python
[FAILED] Traceback detected in simulation_tutorial_2.py
=== stdout ===

=== stderr ===
/usr/lib/python3/dist-packages/jinja2/utils.py:485: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  from collections import MutableMapping
/usr/lib/python3/dist-packages/jinja2/runtime.py:318: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  from collections import Mapping
INFO:SimpleCommunity:I am peer 0, I know 0 peers...
INFO:SimpleCommunity:I am peer 1, I know 0 peers...
INFO:SimpleCommunity:I am peer 2, I know 0 peers...
INFO:SimpleCommunity:I am peer 3, I know 0 peers...
INFO:SimpleCommunity:I am peer 4, I know 0 peers...
INFO:SimpleCommunity:I am peer 5, I know 0 peers...
INFO:SimpleCommunity:I am peer 6, I know 0 peers...
INFO:SimpleCommunity:I am peer 7, I know 0 peers...
INFO:SimpleCommunity:I am peer 8, I know 0 peers...
INFO:SimpleCommunity:I am peer 9, I know 0 peers...
INFO:SimpleCommunity:I am peer 10, I know 0 peers...
INFO:SimpleCommunity:I am peer 11, I know 0 peers...
INFO:SimpleCommunity:I am peer 12, I know 0 peers...
INFO:SimpleCommunity:I am peer 13, I know 0 peers...
INFO:SimpleCommunity:I am peer 14, I know 0 peers...
INFO:SimpleCommunity:I am peer 15, I know 0 peers...
INFO:SimpleCommunity:I am peer 16, I know 0 peers...
INFO:SimpleCommunity:I am peer 17, I know 0 peers...
INFO:SimpleCommunity:I am peer 18, I know 0 peers...
INFO:SimpleCommunity:I am peer 19, I know 0 peers...
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/usr/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "./validate_examples.py", line 35, in validate_run
    exec(compile(module_contents, module_path, 'exec', dont_inherit=True, optimize=0), {})  # pylint: disable=W0122
  File "/home/jenkins/ci/workspace/ipv8/validate_documentation/pyipv8/doc/simulation/simulation_tutorial_2.py", line 69, in <module>
    loop.run_forever()
  File "/home/jenkins/ci/workspace/ipv8/validate_documentation/pyipv8/simulation/discrete_loop.py", line 41, in run_forever
    raise self._exc
  File "/usr/lib/python3.7/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/home/jenkins/ci/workspace/ipv8/validate_documentation/pyipv8/ipv8/test/mocking/endpoint.py", line 40, in send
    raise AssertionError("Received data from unregistered address %s" % repr(socket_address))
AssertionError: Received data from unregistered address UDPv4Address(ip='145.94.62.182', port=45803)
```

--- a/doc/simulation/simulation_tutorial_2.py
+++ b/doc/simulation/simulation_tutorial_2.py
@@ -35,6 +35,7 @@ async def start_communities():
 
         # We assume all peers in our simulation have a public WAN IP (to avoid conflicts with our SimulationEndpoint)
         bootstrap_overlay.my_estimated_lan = ("0.0.0.0", 0)
+        bootstrap_overlay.my_estimated_wan = bootstrap_endpoint.wan_address
         bootstrap_ips.append(bootstrap_endpoint.wan_address)
 
     instances = []
@@ -51,6 +52,7 @@ async def start_communities():
                         extra_communities={'SimpleCommunity': SimpleCommunity})
         # We assume all peers in our simulation have a public WAN IP (to avoid conflicts with our SimulationEndpoint)
         instance.overlays[0].my_estimated_lan = ("0.0.0.0", 0)
+        instance.overlays[0].my_estimated_wan = endpoint.wan_address
         await instance.start()
         instances.append(instance)
 

--- a/ipv8/test/mocking/endpoint.py
+++ b/ipv8/test/mocking/endpoint.py
@@ -51,13 +51,24 @@ class MockEndpoint(Endpoint):
 
 class AddressTester(EndpointListener):
 
+    singleton = None
+
+    def __init__(self, endpoint):
+        super().__init__(endpoint, True)
+        self._get_lan_address(True)
+        AddressTester.singleton = self
+
+    @classmethod
+    def get_singleton(cls, endpoint):
+        if cls.singleton is not None:
+            return cls.singleton
+        return AddressTester(endpoint)
+
     def on_packet(self, packet):
         pass
 
-    def is_lan(self, address):
-        if isinstance(address, UDPv4Address):
-            return self._address_is_lan_without_netifaces(address)
-        return False
+    def is_lan(self, address: str):
+        return self.address_is_lan(address)
 
 
 class AutoMockEndpoint(MockEndpoint):
@@ -98,7 +109,7 @@ class AutoMockEndpoint(MockEndpoint):
         Avoid false positives for the actual machine's lan.
         """
         self._port = address[1]
-        address_tester = AddressTester(self)
+        address_tester = AddressTester.get_singleton(self)
         return address_tester.is_lan(address[0])
 
     def _generate_unique_address(self):

